projects/fishtornado_xr/FishTornado.cpp

```diff
@@ -1229,3 +1256,63 @@ void FishTornadoApp::DrawGui()
         ImGui::EndDisabled();
     }
 }
+
+void FishTornadoApp::SetupMetrics()
+{
+    if (!mSettings.outputMetrics) {
+        return;
+    }
+
+    auto* run = mMetricsData.manager.AddRun("FishTornado Metrics");
+
+    mMetricsData.pGpuFrameTimeGauge = run->AddMetric<ppx::metrics::MetricGauge>({"GPU Frame Time", "ms", ppx::metrics::MetricInterpretation::LOWER_IS_BETTER, {0.f, 60000.f}});
+    mMetricsData.pCpuFrameTimeGauge = run->AddMetric<ppx::metrics::MetricGauge>({"Total (CPU) Frame Time", "ms", ppx::metrics::MetricInterpretation::LOWER_IS_BETTER, {0.f, 60000.f}});
+    mMetricsData.pIAVertGauge       = run->AddMetric<ppx::metrics::MetricGauge>({"IA Vertices", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+    mMetricsData.pIAPrimGauge       = run->AddMetric<ppx::metrics::MetricGauge>({"IA Primitives", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+    mMetricsData.pVSInvGauge        = run->AddMetric<ppx::metrics::MetricGauge>({"VS Invocations", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+    mMetricsData.pCInvGauge         = run->AddMetric<ppx::metrics::MetricGauge>({"C Invocations", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+    mMetricsData.pCPrimGauge        = run->AddMetric<ppx::metrics::MetricGauge>({"C Primitives", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+    mMetricsData.pPSInvGauge        = run->AddMetric<ppx::metrics::MetricGauge>({"PS Invocations", "", ppx::metrics::MetricInterpretation::NONE, {0.f, 1000000000.f}});
+
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pGpuFrameTimeGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pCpuFrameTimeGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pIAVertGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pIAPrimGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pVSInvGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pCInvGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pCPrimGauge);
+    mMetricsData.allMetrics.emplace_back(mMetricsData.pPSInvGauge);
+}
+
+void FishTornadoApp::WriteMetrics()
+{
+    if (!mSettings.outputMetrics) {
+        return;
+    }
+
+    auto            validPath = ppx::fs::GetValidPathToOutputFile(std::filesystem::path(kMetricsFilename));
+    ppx::CSVFileLog metricsFileLog(validPath.c_str());
```

Comment:
So, Fishtornado is NOT in the ppx namespace (and we avoid the 'using' keyword). That's why ppx:: is all over the file.

Reply:
Is it? we use `grfx::` directly which is in the `ppx` namespace. Also, removing the `ppx::` prefix doesn't causes issues?
(`using` is already used in the headers)

Reply:
Oh, ew. It's opaque. You're right, it's defined in some headers (ocean, flocking, etc.) and not others. It's not in the main file, but it's leaking into it because of that.

Anyway, I don't think this is a change for this PR. I'm following the pattern largely defined in the file for now; I do think we should make a follow-up that cleans things up in one direction or the other. My preference would be to ditch 'using namespace ppx' across all projects to avoid opaquely leaking namespaces and creating aliases.